node/src/gql/mutation/me/mentor-keywords-update.ts

```diff
@@ -0,0 +1,98 @@
+
+The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+*/
+import {
+  GraphQLObjectType,
+  GraphQLBoolean,
+  GraphQLID,
+  GraphQLInputObjectType,
+  GraphQLList,
+  GraphQLString,
+} from 'graphql';
+import {
+  Mentor as MentorModel,
+  Keyword as KeywordModel,
+} from '../../../models';
+import { Mentor } from '../../../models/Mentor';
+import { User, UserRole } from '../../../models/User';
+
+export interface UpdateKeyword {
+  name: string;
+  type: string;
+}
+
+export const UpdateKeywordType = new GraphQLInputObjectType({
+  name: 'UpdateKeywordType',
+  fields: () => ({
+    name: { type: GraphQLString },
+    type: { type: GraphQLString },
+  }),
+});
+
+export const updateMentorKeywords = {
+  type: GraphQLBoolean,
+  args: {
+    mentorId: { type: GraphQLID },
+    keywords: { type: GraphQLList(UpdateKeywordType) },
+  },
+  resolve: async (
+    _root: GraphQLObjectType,
+    args: { mentorId: string; keywords: UpdateKeyword[] },
+    context: { user: User }
+  ): Promise<boolean> => {
+    let mentor: Mentor = await MentorModel.findOne({
+      user: context.user._id,
+    });
+    // Check mentor permissions
+    if (!mentor) {
+      throw new Error('you do not have a mentor');
+    }
+    if (args.mentorId && `${mentor._id}` !== `${args.mentorId}`) {
+      if (
+        context.user.userRole !== UserRole.ADMIN &&
+        context.user.userRole !== UserRole.CONTENT_MANAGER
+      ) {
+        throw new Error('you do not have permission to edit this mentor');
+      }
+      mentor = await MentorModel.findById(args.mentorId);
+    }
+
+    // Batch write all keywords
+    if (args.keywords && args.keywords.length > 0) {
+      await KeywordModel.bulkWrite(
+        args.keywords.map((k) => {
+          return {
+            updateOne: {
+              filter: { name: k.name },
+              update: k,
+              upsert: true,
+            },
+          };
+        })
+      );
+    }
+
+    // Update mentor keywords
+    const keywords = await KeywordModel.find({
+      name: { $in: args.keywords.map((k) => k.name) },
+    });
+    const updated = await MentorModel.findByIdAndUpdate(
+      mentor._id,
+      {
+        $set: {
+          keywords: keywords.map((k) => k._id),
+        },
+      },
+      {
+        new: true,
+        upsert: true,
```

Comment:
Probably don't want to upsert here just on the off chance that a mentor doesn't exists, it'd create an empty mentor with just keywords.